### DISCUSSION TITLE
backward compatible for spawn_client in 1.0.0 release

### DISCUSSION THREAD

**Discussion by xliao-neology** [actor-framework/actor-framework#1899] [Q&A]
we been using spawn_client() perfectly in 0.19.6. However same code will have below error show up in 1.0.0.
I understand the trend is to use caf_net. But for now, any way that I can do to keep spawn_client() work with 1.0.0 release?

>>>>>
/usr/local/include/caf/actor_storage.hpp:30:5: error: ‘caf::io::abstract_broker::abstract_broker(caf::actor_config&)’ is protected within this context
   30 |     new (&data) T(std::forward<Us>(zs)...);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

**Comment by Neverlord** [comment]
Is this happening without any code changes on your end?

**Reply by xliao-neology** [reply]
yes. The only change is compiling and using 1.0.0 CAF library.
More details compile errors as below. I check /usr/local/include/caf/io/abstract_broker.hpp,  not much changed at all. I doubt might be other places' code refractor causing this. Looks like a bug for me .

>>>>
In file included from /usr/local/include/caf/make_actor.hpp:7,
                 from /usr/local/include/caf/actor_system.hpp:21,
                 from /usr/local/include/caf/actor_from_state.hpp:7:
/usr/local/include/caf/actor_storage.hpp: In instantiation of ‘caf::actor_storage<T>::actor_storage(caf::actor_id, caf::node_id, caf::actor_system*, Us&& ...) [with Us = {caf::actor_config&}; T = caf::io::broker; caf::actor_id = long unsigned int]’:
/usr/local/include/caf/make_actor.hpp:33:14:   required from ‘R caf::make_actor(actor_id, node_id, actor_system*, Ts&& ...) [with T = io::broker; R = actor; Ts = {actor_config&}; actor_id = long unsigned int]’
/usr/local/include/caf/actor_system.hpp:519:29:   required from ‘caf::infer_handle_from_class_t<T> caf::actor_system::spawn_impl(caf::actor_config&, Ts&& ...) [with C = caf::io::broker; caf::spawn_options Os = caf::spawn_options::no_flags; Ts = {}; caf::infer_handle_from_class_t<T> = caf::actor]’
/usr/local/include/caf/actor_system.hpp:375:29:   required from ‘caf::infer_handle_from_class_t<T> caf::actor_system::spawn_class(caf::actor_config&, Ts&& ...) [with C = caf::io::broker; caf::spawn_options Os = caf::spawn_options::no_flags; Ts = {}; caf::infer_handle_from_class_t<T> = caf::actor]’
/usr/local/include/caf/io/middleman.hpp:293:42:   required from ‘caf::expected<typename caf::infer_handle_from_class<Impl, is_base_of_v<caf::abstract_actor, Impl> >::type> caf::io::middleman::spawn_client_impl(F, const std::string&, uint16_t, Ts&& ...) [with caf::spawn_options Os = caf::spawn_options::no_flags; Impl = caf::io::broker; F = caf::behavior (*)(caf::io::broker*, caf::io::connection_handle, const caf::actor&); Ts = {neo_tzc::idris::idrisSubUnit_impl*}; typename caf::infer_handle_from_class<Impl, is_base_of_v<caf::abstract_actor, Impl> >::type = caf::actor; std::string = std::__cxx11::basic_string<char>; uint16_t = short unsigned int]’
/usr/local/include/caf/io/middleman.hpp:222:39:   required from ‘caf::expected<typename caf::infer_handle_from_fun_trait<typename caf::detail::get_callable_trait<F>::type>::type> caf::io::middleman::spawn_client(F, const std::string&, uint16_t, Ts&& ...) [with caf::spawn_options Os = caf::spawn_options::no_flags; F = caf::behavior (*)(caf::io::broker*, caf::io::connection_handle, const caf::actor&); Ts = {neo_tzc::idris::idrisSubUnit_impl*}; typename caf::infer_handle_from_fun_trait<typename caf::detail::get_callable_trait<F>::type>::type = caf::actor; typename caf::detail::get_callable_trait<F>::type = caf::detail::callable_trait<caf::behavior (*)(caf::io::broker*, caf::io::connection_handle, const caf::actor&)>; std::string = std::__cxx11::basic_string<char>; uint16_t = short unsigned int]’

/usr/local/include/caf/actor_storage.hpp:30:5: error: ‘caf::io::abstract_broker::abstract_broker(caf::actor_config&)’ is protected within this context
   30 |     new (&data) T(std::forward<Us>(zs)...);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/local/include/caf/io/basp/routing_table.hpp:7,
                 from /usr/local/include/caf/io/basp/instance.hpp:11,
                 from /usr/local/include/caf/io/basp/all.hpp:10,
                 from /usr/local/include/caf/io/all.hpp:7,
/usr/local/include/caf/io/abstract_broker.hpp:360:12: note: declared protected here
  360 |   explicit abstract_broker(actor_config& cfg);

**Reply by Neverlord** [reply]
Yeah, this is bug. Moving to issues. Sorry for the inconvenience, we'll look into it!

**Comment by Neverlord** [comment]
See https://github.com/actor-framework/actor-framework/issues/1900
